modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java

```diff
@@ -665,6 +673,30 @@ private void _validate(
 		}
 	}
 
+	private void _validateExternalReferenceCode(
+			String externalReferenceCode, long commerceCurrencyId,
+			long companyId)
+		throws DuplicateCommerceCurrencyExternalReferenceCodeException {
+
+		if (Validator.isNull(externalReferenceCode)) {
+			return;
+		}
+
+		CommerceCurrency commerceCurrency =
+			commerceCurrencyPersistence.fetchByERC_C(
+				externalReferenceCode, companyId);
+
+		if (commerceCurrency == null) {
+			return;
+		}
+
+		if (commerceCurrency.getCommerceCurrencyId() != commerceCurrencyId) {
+			throw new DuplicateCommerceCurrencyExternalReferenceCodeException(
+				"There is another commerce currency with external reference " +
+					"code " + externalReferenceCode);
+		}
+	}
+
```

Comment:
there is no need to implement a validator for duplication control because it is automatically generated see [here](https://github.com/victorcdp/liferay-portal/blob/7f63ecc3e6b2c30122aea51cb9cbf2649c11ea62/modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/persistence/impl/CommerceCurrencyPersistenceImpl.java#L4098)